core/embed/rust/build.rs

```diff
@@ -14,12 +14,54 @@ fn main() {
     link_core_objects();
 }
 
-// fn block_words() -> String {
-//     match env::var("FLASH_BLOCK_WORDS") {
-//         Ok(model) => model,
-//         Err(_) => panic!("FLASH_BLOCK_WORDS not set")
-//     }
-// }
+const DEFAULT_BINDGEN_MACROS_COMMON: &[&str] = &[
+    "-I../unix",
+    "-I../trezorhal/unix",
+    "-I../../build/unix",
+    "-I../../vendor/micropython/ports/unix",
+    "-I../../../crypto",
+    "-I../../../storage",
+    "-I../../vendor/micropython",
+    "-I../../vendor/micropython/lib/uzlib",
+    "-I../lib",
+    "-I../trezorhal",
+    "-I../trezorhal/unix",
+    "-I../models",
+    "-DTREZOR_EMULATOR",
+    "-DTREZOR_BOARD=\"boards/board-unix.h\"",
+];
+
+#[cfg(feature = "model_tt")]
+const DEFAULT_BINDGEN_MACROS_T2T1: &[&str] = &[
+    "-DSTM32F427",
+    "-DTREZOR_MODEL_T",
+    "-DFLASH_BIT_ACCESS=1",
+    "-DFLASH_BLOCK_WORDS=1",
+];
+#[cfg(not(feature = "model_tt"))]
+const DEFAULT_BINDGEN_MACROS_T2T1: &[&str] = &[];
+
+#[cfg(feature = "model_tr")]
+const DEFAULT_BINDGEN_MACROS_T2B1: &[&str] = &[
+    "-DSTM32F427",
+    "-DTREZOR_MODEL_R",
+    "-DFLASH_BIT_ACCESS=1",
+    "-DFLASH_BLOCK_WORDS=1",
+];
+#[cfg(not(feature = "model_tr"))]
+const DEFAULT_BINDGEN_MACROS_T2B1: &[&str] = &[];
+
+fn add_bindgen_macros<'a>(clang_args: &mut Vec<&'a str>, envvar: Option<&'a str>) {
+    let default_macros = DEFAULT_BINDGEN_MACROS_COMMON
+        .iter()
+        .chain(DEFAULT_BINDGEN_MACROS_T2T1)
+        .chain(DEFAULT_BINDGEN_MACROS_T2B1);
```

Comment:
Don't we miss also `T3T1` specifics?

Reply:
ah, this is a cherrypick of https://github.com/trezor/trezor-firmware/commit/cf58fdd31306f4560b357ec9980d61026d44d878 from main. we probably should, but this commit should be removed from the PR before merging, and someone should add T3T1 features when merging the UI branch into main (or later, whenever we hit a problem caused by it, so probably when you try to run T3T1 Rust tests)